### ISSUE TITLE
Sorting by last modified date doesn't always return the correct sorting order

### ISSUE THREAD

**Issue by sadpandajoe** [apache/superset#27500]
### Bug description

When viewing charts, if you try to sort by last modified, not all dates returned are sorted in order.

### How to reproduce the bug

1. Go to the chart list page making sure you have charts that have different modified dates so they don't all belong to the same day 
2. Click on the sort by last modified in the header
3. Notice in the column that some of the last modified dates are out of order.

### Screenshots/recordings

![Screenshot 2024-03-13 at 10 12 28 AM](https://github.com/apache/superset/assets/47334392/3c8d4076-9745-4fff-9725-68bdc34ea213)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by michael-s-molina** [comment]
@sadpandajoe I think this a problem with our example charts that are loaded with `NULL` values for the `last_saved_at` column. Are you able to reproduce this using non-example charts?

**Comment by sadpandajoe** [comment]
> @sadpandajoe I think this a problem with our example charts that are loaded with `NULL` values for the `last_saved_at` column. Are you able to reproduce this using non-example charts?

Yeah this was part of our prod instance that I threw the head of 4.0 branch onto.

**Comment by rusackas** [comment]
It seems like this is also happening with non-example charts, even on the latest `master` or Preset. @sadpandajoe hit me up if you want to take a look more synchronously.

**Comment by rusackas** [comment]
Can anyone confirm this is still happening in 4.1.2?

**Comment by rusackas** [comment]
@sadpandajoe this still happening?

**Comment by rusackas** [comment]
Is the incorrect last-modified date sorting still happening in recent versions? @rusackas asked @sadpandajoe in June 2025 with no response. Please confirm or this will be closed as inactive.
